docs/decisions/managing-service-dependencies.md

```diff
@@ -0,0 +1,86 @@
+# Managing Service Dependencies
+
+## Context and Problem Statement
+
+As we split OpenFoodFacts up into smaller, more manageable, services, it becomes increasingly difficult for developers to know what additional services need to be deployed in order for the service they are working on to function correctly. This becomes increasingly difficult if the referenced service itself has other dependencies.
+
+Note that for development and testing purposes it is recommended that any dependencies are mocked or managed using tools like testcontainers, rather than relying on a separate service being up. This document concerns itself more with creating a workable end-to-end deployment that can be used for ad-hoc, interactive testing and discovery.
+
+This document is specifically concerned with "internal" dependencies where shared state is needed, e.g. each service can have its own independent database rather than relying on a shared database service, but the messaging system (Redis) needs to be shared in order for messages to pass between services. Dependencies on services outside the scope of OpenFoodFacts are referred to as "external" dependencies.
```

Comment:
```suggestion
This document is specifically concerned with "internal" dependencies where shared state is needed, e.g. each service can have its own independent database rather than relying on a shared database service, but the messaging system (Redis) needs to be shared in order for messages to pass between services. Dependencies on services outside the scope of Open Food Facts are referred to as "external" dependencies.
```